torch/csrc/jit/codegen/cuda/expr_simplifier.cpp

```diff
@@ -413,14 +469,90 @@ Val* unflatten(Val* value, const std::list<ValInfo>& variables) {
     }
     return lhs;
   }
+  return value;
+}
+
+Val* unflatten(Val* value, const std::list<ValInfo>& variables) {
+  using namespace std::placeholders;
+  return recurseDown(value, std::bind(unflattenRule, _1, variables));
+}
+
+} // namespace assoc_comm
+
+namespace rules {
+
+using FOp = assoc_comm::FlattenedAssocCommOp;
+
+// Do simplifications like:
+// 1 * a -> a
+// 0 * a -> 0
+// true && a -> a
+// false && a -> false
+// true || a -> true
+// false || a -> a
+// a % 1 -> 0
+// a / 1 -> a
+// ...
+Val* eliminateTrivialComputation(Val* value) {
+  if (auto fop = dynamic_cast<FOp*>(value->definition())) {
+    if (fop->isTrivial()) {
+      return fop->input(0);
+    }
+    if (fop->hasConstantTerm()) {
+      auto const_term = fop->getConstantTerm();
+      auto op = fop->getOpType();
+      if (assoc_comm::isIdentity(const_term, op)) {
+        // 1 * a -> a
+        if (fop->inputs().size() == 1) {
+          return fop->input(0);
+        } else {
+          auto output = IrBuilder::newScalar(*value->getDataType());
+          IrBuilder::create<FOp>(op, output, fop->inputs());
+          return output;
+        }
+      } else if (assoc_comm::isBlackhole(const_term, op)) {
+        // 0 * a -> 0
+        return const_term;
+      }
+    }
+  } else if (auto bop = dynamic_cast<BinaryOp*>(value->definition())) {
+    if (bop->getBinaryOpType() == BinaryOpType::Mod) {
+      // a % 1 -> 0
+      auto rhs = foldConstants(bop->rhs());
+      if (rhs->isOneInt()) {
+        return IrBuilder::newConstant(0, *value->getDataType());
+      }
+    } else if (
+        bop->getBinaryOpType() == BinaryOpType::Div ||
+        bop->getBinaryOpType() == BinaryOpType::CeilDiv) {
+      // a / 1 -> a
+      // 0 / a -> 0
```

Comment:
Does `0.0 / 0.0` results in `0.0` or produce a zero-div exception?

Reply:
Ahh, good question. If `a` is symbolic whose runtime value can be zero, then this will simplify it to zero, and during runtime, the zero-division error will be lost. If `a` is a zero constant, then before running this pass, `foldConstants` will be called, which will throw a zero division error.

Reply:
Probably I should be safer, and only enable this if I can prove that `a` is not zero? I think this makes more sense.

Reply:
changed